iocore/dns/DNS.cc

```diff
@@ -424,12 +430,27 @@ DNSEntry::init(const char *x, int len, int qtype_arg, Continuation *acont, DNSPr
 }
 
 /**
+ Open UDP and/or TCP connections based on dns_conn_mode
+ */
+
+void
+DNSHandler::open_cons(sockaddr const *target, bool failed, int icon)
+{
+  if (dns_conn_mode != DNS_CONN_MODE::TCP_ONLY) {
+    open_con(target, failed, icon, false);
+  }
+  if (dns_conn_mode != DNS_CONN_MODE::UDP_ONLY) {
```

Comment:
With the setting of "1", does this imply that we always create the TCP connection, regardless if it's needed or not? That seems a little wasteful, no? Or am I not reading this right ?